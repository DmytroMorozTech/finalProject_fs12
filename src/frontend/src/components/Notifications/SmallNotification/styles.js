import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  smallNotification: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
    border: theme.border.simple,
    borderRadius: theme.shape.medium,
    backgroundColor: theme.palette.common.white,
    padding: theme.spacing(2),
    marginBottom: theme.spacing(1)
  },

  item: {
    display: 'flex',
    alignItems: 'center'
  },

  block: {
    marginTop: theme.spacing(1),
    marginBottom: theme.spacing(1)
  },

  icon: {
    color: theme.palette.grey[600],
    fontSize: theme.typography.icons.medium.fontSize
  },

  title: {
    fontSize: theme.typography.h5.fontSize,
    fontWeight: theme.typography.fontWeightLight,
    color: theme.palette.grey[600],
    paddingLeft: theme.spacing(2)
  },

  link: {
    textDecoration: 'none',
    color: theme.palette.primary.main,
    fontWeight: theme.typography.fontWeightBold,
    fontSize: theme.typography.h5.fontSize,
    cursor: 'pointer'
  }
}))
