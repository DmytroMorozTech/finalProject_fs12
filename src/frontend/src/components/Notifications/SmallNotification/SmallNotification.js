import NotificationsNoneOutlinedIcon from '@material-ui/icons/NotificationsNoneOutlined'
import React from 'react'
import styles from './styles'
import { useDispatch } from 'react-redux'
import { markAllNotificationsAsViewedAction } from '../../../redux/Notification/notificationActions'

function SmallNotification () {
  const classes = styles()

  const dispatch = useDispatch()

  return (
    <div className={classes.smallNotification}>
      <div className={classes.item}>
        <NotificationsNoneOutlinedIcon fontSize="inherit" className={classes.icon}/>
        <span className={classes.title}>Your notifications live here</span>
      </div>
      <div
        id='markAllNotificationsAsViewed'
        onClick={() => dispatch(markAllNotificationsAsViewedAction())}
        className={classes.link}>
        Mark all as viewed
      </div>
    </div>
  )
}

export default SmallNotification
