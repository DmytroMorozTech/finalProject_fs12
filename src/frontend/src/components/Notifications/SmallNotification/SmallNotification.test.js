import { shallow } from 'enzyme'
import * as redux from 'react-redux'
import React from 'react'
import SmallNotification from './SmallNotification'

jest.mock('./styles', () => () => ({}))

const setUp = () => shallow(<SmallNotification />)

describe('should render SmallNotification component', () => {
  let spyOnUseSelector
  let spyOnUseDispatch
  let mockDispatch

  let notification

  let component

  beforeEach(() => {
    notification = [{'id': 127, 'createdDate': '2021-10-19T20:48:14.466256', 'type': 'NEW_POST_WAS_CREATED', 'data': {'postId': 43}, 'userWhoTriggered': {'imgPublicId': 'linkedin/avatars/vrhpzgx8di9aoz5hykzh', 'fullName': 'Jamie Anderson', 'id': '11', 'text': 'sdfsdfsdf...'}, 'isViewed': false}, {'id': 117, 'createdDate': '2021-10-19T20:48:11.266651', 'type': 'NEW_POST_WAS_CREATED', 'data': {'postId': 42}, 'userWhoTriggered': {'imgPublicId': 'linkedin/avatars/vrhpzgx8di9aoz5hykzh', 'fullName': 'Jamie Anderson', 'id': '11', 'text': 'sdlfksdlkfj...'}, 'isViewed': false}, {'id': 103, 'createdDate': '2021-10-19T20:34:14.712304', 'type': 'NEW_POST_WAS_CREATED', 'data': {'postId': 41}, 'userWhoTriggered': {'imgPublicId': 'linkedin/avatars/vrhpzgx8di9aoz5hykzh', 'fullName': 'Jamie Anderson', 'id': '11', 'text': 'sdfsdfsdf...'}, 'isViewed': false}, {'id': 93, 'createdDate': '2021-10-19T20:34:12.472196', 'type': 'NEW_POST_WAS_CREATED', 'data': {'postId': 40}, 'userWhoTriggered': {'imgPublicId': 'linkedin/avatars/vrhpzgx8di9aoz5hykzh', 'fullName': 'Jamie Anderson', 'id': '11', 'text': 'sdfsdfsdf...'}, 'isViewed': false}, {'id': 83, 'createdDate': '2021-10-19T20:33:59.256914', 'type': 'NEW_POST_WAS_CREATED', 'data': {'postId': 39}, 'userWhoTriggered': {'imgPublicId': 'linkedin/avatars/vrhpzgx8di9aoz5hykzh', 'fullName': 'Jamie Anderson', 'id': '11', 'text': 'sdfslkdjfksjdfksjdfh\nksadjfksdjfh\n\nskdjfhskdjfh\n\n\nosdfjlsdkfj...'}, 'isViewed': false}, {'id': 58, 'createdDate': '2021-10-19T20:03:07.023922', 'type': 'NEW_POST_WAS_CREATED', 'data': {'postId': 32}, 'userWhoTriggered': {'imgPublicId': 'linkedin/avatars/vrhpzgx8di9aoz5hykzh', 'fullName': 'Jamie Anderson', 'id': '11', 'text': 'new post...'}, 'isViewed': false}, {'id': 48, 'createdDate': '2021-10-19T20:00:02.337555', 'type': 'NEW_POST_WAS_CREATED', 'data': {'postId': 31}, 'userWhoTriggered': {'imgPublicId': 'linkedin/avatars/vrhpzgx8di9aoz5hykzh', 'fullName': 'Jamie Anderson', 'id': '11', 'text': 'sdf...'}, 'isViewed': false}, {'id': 38, 'createdDate': '2021-10-19T19:59:41.036487', 'type': 'NEW_POST_WAS_CREATED', 'data': {'postId': 30}, 'userWhoTriggered': {'imgPublicId': 'linkedin/avatars/vrhpzgx8di9aoz5hykzh', 'fullName': 'Jamie Anderson', 'id': '11', 'text': 'new post...'}, 'isViewed': false}]

    spyOnUseSelector = jest.spyOn(redux, 'useSelector')
    spyOnUseSelector.mockReturnValue(notification)
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    component = setUp()
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
  it('should work markAllNotificationsAsViewed dispatch', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0)
    component.find('SharedButton[id=\'markAllNotificationsAsViewed\']').simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
})
