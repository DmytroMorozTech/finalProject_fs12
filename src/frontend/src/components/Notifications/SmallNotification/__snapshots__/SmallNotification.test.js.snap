// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SmallNotification component to match snapshot 1`] = `
<div>
  <div>
    <Memo(ForwardRef(NotificationsNoneOutlinedIcon))
      fontSize="inherit"
    />
    <span>
      Your notifications live here
    </span>
  </div>
  <div
    id="markAllNotificationsAsViewed"
    onClick={[Function]}
  >
    Mark all as viewed
  </div>
</div>
`;
